application/client.core/src/app/environment/components/sidebar/adb/logcat/component.ts

```diff
@@ -118,10 +170,53 @@ export class SidebarAppAdbLogcatComponent implements OnInit, OnDestroy {
         });
     }
 
+    public _ng_onFocus() {
+        this._prevProcessSelected = this._ng_processSelected;
+        this._ng_processSearchTerm = '';
+        this._processSearchTerm.next(this._ng_processSearchTerm);
+    }
+
+    public _ng_onBlur() {
+        this._ng_processSelected = this._prevProcessSelected;
+        this._ng_processSearchTerm = '';
+        this._processSearchTerm.next(this._ng_processSearchTerm);
+    }
+
+    public _ng_onCloseSearch() {
+        this._ng_processSearchTerm = '';
+        this._processSearchTerm.next(this._ng_processSearchTerm);
+    }
+
+    public _ng_onKeyup(event: KeyboardEvent) {
+        if (event.key === 'Escape') {
+            this._ng_processSearchTerm = '';
+        } else if (event.key === 'Backspace') {
+            this._ng_processSearchTerm = this._ng_processSearchTerm.slice(0, this._ng_processSearchTerm.length - 1);
+        } else if (event.key.length > 1) {
+            return;
+        } else {
+            this._ng_processSearchTerm += event.key;
+        }
+        this._processSearchTerm.next(this._ng_processSearchTerm);
+    }
+
+    public _ng_getSafeHTML(str: string): SafeHtml {
+        return this._sanitizer.bypassSecurityTrustHtml(str);
+    }
+
+    private _filter(value: string): IPair[] {
+        if (typeof value !== 'string') {
+            return;
+        }
+        const scored = sortPairs(this._processPairs, value, value !== '', 'span');
+        return scored;
+    }
+
     private _init() {
         this.service.status = EAdbStatus.init;
         this._ng_devices = [this._ng_noDevice];
         this._ng_processes = [this._ng_noProcess];
+        this._processPairs = [];
         this._ng_amount = this.service.bytesToString(0);
         this._ng_logLevel = this._ng_logLevels[0].flag;
         this._ng_device = this._ng_devices[0];
```

Comment:
That shouldn't be possible, because `this._ng_devices` is always initialized with at least `this._ng_noDevice`
Same goes for all the other ones like `this._ng_logLevel` or `this._ng_processes`